btree/btree.c

```diff
@@ -3469,19 +3479,17 @@ static int64_t btree_put_kv_tick(struct m0_sm_op *smop)
 			return P_LOCK;
 		else
 			return P_SETUP;
-	case P_SETUP: {
+	case P_LOCKALL:
+		M0_ASSERT(bop->bo_flags & BOF_LOCKALL);
+		return lock_op_init(&bop->bo_op, &bop->bo_i->i_nop,
+				    bop->bo_arbor->t_desc, P_SETUP);
+	case P_SETUP:
 		oi->i_height = tree->t_height;
 		level_alloc(oi, oi->i_height);
 		if (oi->i_level == NULL)
 			return fail(bop, M0_ERR(-ENOMEM));
```

Comment:
If lock_acquired flag is set and oi->i_level == NULL, unlock the tree lock at line 3490and then return fail()

Reply:
Done.